pycue/opencue/cuebot.py

```diff
@@ -265,3 +281,95 @@ def getStub(cls, name):
     @staticmethod
     def getConfig():
         return config
+
+
+# Python 2/3 compatible implementation of ABC
+ABC = abc.ABCMeta('ABC', (object,), {'__slots__': ()})
+
+
+class SleepingPolicy(ABC):
+    """
+    Implement policy for sleeping between API retries
+    """
+    @abc.abstractmethod
+    def sleep(self, attempt):
+        """
+        How long to sleep in milliseconds.
+        :param attempt: the number of attempt (starting from zero)
+        """
+        assert attempt >= 0
+
+
+class ExponentialBackoff(SleepingPolicy):
+    """
+    Implement policy that will increase retry period by exponentially in every try
+    """
+    def __init__(self,
+                 init_backoff_ms,
+                 max_backoff_ms,
+                 multiplier=2):
+        """
+        inputs in ms
+        """
+        self._init_backoff = init_backoff_ms
+        self._max_backoff = max_backoff_ms
+        self._multiplier = multiplier
+
+    def sleep(self, attempt):
+        sleep_time_ms = min(
+            self._init_backoff * self._multiplier ** attempt,
+            self._max_backoff
+        )
+        time.sleep(sleep_time_ms / 1000)
```

Comment:
should we make the second value a float so it works the same across python2/3?

python 2
```
>>> 1600 / 1000
1
```

python3
```
>>> 1600 / 1000                                                                                                                                                                                                
1.6
```

Reply:
Good catch